### ISSUE TITLE
Issue: MadPlotLib and cv2 not found in windows standalone

### ISSUE THREAD

**Issue by fambaa** [Fannovel16/comfy_controlnet_preprocessors#6]
I got this huge wall of text saying that all requirements are satisfied, yet it could not find cv2.

I then did a normal install of comfy, and not the standalone version went through the install procedure there and redid your controlnet node install as instructed and no problem, now the nodes show up.

I am a python noob, so I have no clue what causes this issue. Maybe I did something wrong earlier.  Here is the wall of text where you can see this contradicting issue. Probably has something to do with the python standalone version.

` 
D:\ComfyUI_windows_portable>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build --dont-upcast-attention
disabling upcasting of attention
Set vram state to: NORMAL VRAM
Using xformers cross attention
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\canny\__init__.py", line 1, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\canny module for custom nodes: No module named 'cv2'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 879, in exec_module
  File "<frozen importlib._bootstrap_external>", line 1016, in get_code
  File "<frozen importlib._bootstrap_external>", line 1073, in get_data
FileNotFoundError: [Errno 2] No such file or directory: 'D:\\ComfyUI_windows_portable\\ComfyUI\\custom_nodes\\ckpts\\__init__.py'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\ckpts module for custom nodes: [Errno 2] No such file or directory: 'D:\\ComfyUI_windows_portable\\ComfyUI\\custom_nodes\\ckpts\\__init__.py'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_controlnet_preprocessors\__init__.py", line 1, in <module>
    from . import canny, hed, midas, mlsd, openpose, uniformer
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_controlnet_preprocessors\canny\__init__.py", line 1, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_controlnet_preprocessors module for custom nodes: No module named 'cv2'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\hed\__init__.py", line 2, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\hed module for custom nodes: No module named 'cv2'
Installing requirements...
Requirement already satisfied: opencv-contrib-python in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 1)) (4.7.0.72)
Requirement already satisfied: opencv-python in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 2)) (4.7.0.72)
Requirement already satisfied: numpy==1.23.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 3)) (1.23.1)
Requirement already satisfied: timm==0.6.12 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (0.6.12)
Requirement already satisfied: torchvision in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 5)) (0.15.1)
Requirement already satisfied: scipy in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 6)) (1.10.1)
Requirement already satisfied: matplotlib in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (3.7.1)
Requirement already satisfied: scikit-image in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 8)) (0.20.0)
Requirement already satisfied: basicsr==1.4.2 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (1.4.2)
Requirement already satisfied: prettytable==3.6.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 10)) (3.6.0)
Requirement already satisfied: svglib in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 11)) (1.5.1)
Requirement already satisfied: reportlab in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from -r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 12)) (3.6.12)
Requirement already satisfied: torch>=1.7 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (2.0.0)
Requirement already satisfied: pyyaml in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (6.0)
Requirement already satisfied: huggingface-hub in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (0.13.2)
Requirement already satisfied: addict in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (2.4.0)
Requirement already satisfied: future in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.18.3)
Requirement already satisfied: lmdb in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (1.4.0)
Requirement already satisfied: Pillow in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (9.4.0)
Requirement already satisfied: requests in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (2.28.2)
Requirement already satisfied: tb-nightly in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (2.13.0a20230316)
Requirement already satisfied: tqdm in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (4.65.0)
Requirement already satisfied: yapf in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.32.0)
Requirement already satisfied: wcwidth in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from prettytable==3.6.0->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 10)) (0.2.6)
Requirement already satisfied: typing-extensions in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from torch>=1.7->timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (4.5.0)
Requirement already satisfied: filelock in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from torch>=1.7->timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (3.10.0)
Requirement already satisfied: sympy in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from torch>=1.7->timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (1.11.1)
Requirement already satisfied: networkx in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from torch>=1.7->timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (3.0)
Requirement already satisfied: jinja2 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from torch>=1.7->timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (3.1.2)
Requirement already satisfied: pyparsing>=2.3.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (2.8.2)
Requirement already satisfied: packaging>=20.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (23.0)
Requirement already satisfied: contourpy>=1.0.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (1.0.7)
Requirement already satisfied: kiwisolver>=1.0.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (1.4.4)
Requirement already satisfied: fonttools>=4.22.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (4.39.2)
Requirement already satisfied: cycler>=0.10 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (0.11.0)
Requirement already satisfied: lazy_loader>=0.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from scikit-image->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 8)) (0.1)
Requirement already satisfied: PyWavelets>=1.1.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from scikit-image->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 8)) (1.4.1)
Requirement already satisfied: imageio>=2.4.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from scikit-image->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 8)) (2.26.0)
Requirement already satisfied: tifffile>=2019.7.26 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from scikit-image->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 8)) (2023.3.15)
Requirement already satisfied: lxml in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from svglib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 11)) (4.9.2)
Requirement already satisfied: tinycss2>=0.6.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from svglib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 11)) (1.2.1)
Requirement already satisfied: cssselect2>=0.2.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from svglib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 11)) (0.7.0)
Requirement already satisfied: webencodings in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from cssselect2>=0.2.0->svglib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 11)) (0.5.1)Requirement already satisfied: six>=1.5 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from python-dateutil>=2.7->matplotlib->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 7)) (1.16.0)
Requirement already satisfied: colorama in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tqdm->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.4.6)
Requirement already satisfied: idna<4,>=2.5 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from requests->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (3.4)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from requests->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (3.1.0)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from requests->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (2022.12.7)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from requests->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (1.26.15)
Requirement already satisfied: wheel>=0.26 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.40.0)Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.7.0)
Requirement already satisfied: werkzeug>=1.0.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (2.2.3)
Requirement already satisfied: grpcio>=1.48.2 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (1.51.3)
Requirement already satisfied: setuptools>=41.0.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (57.4.0)
Requirement already satisfied: protobuf>=3.19.6 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (4.22.1)
Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (1.8.1)
Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.4.6)
Requirement already satisfied: google-auth<3,>=1.6.3 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (2.16.2)
Requirement already satisfied: markdown>=2.6.8 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (3.4.1)
Requirement already satisfied: absl-py>=0.4 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (1.4.0)Requirement already satisfied: cachetools<6.0,>=2.0.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from google-auth<3,>=1.6.3->tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (5.3.0)
Requirement already satisfied: pyasn1-modules>=0.2.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from google-auth<3,>=1.6.3->tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.2.8)
Requirement already satisfied: rsa<5,>=3.1.4 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from google-auth<3,>=1.6.3->tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (4.9)
Requirement already satisfied: requests-oauthlib>=0.7.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (1.3.1)
Requirement already satisfied: MarkupSafe>=2.1.1 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from werkzeug>=1.0.1->tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (2.1.2)
Requirement already satisfied: mpmath>=0.19 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from sympy->torch>=1.7->timm==0.6.12->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 4)) (1.3.0)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (0.4.8)
Requirement already satisfied: oauthlib>=3.0.0 in c:\users\chris\appdata\local\programs\python\python310\lib\site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tb-nightly->basicsr==1.4.2->-r D:\ComfyUI_windows_portable\ComfyUI\custom_nodes/requirements.txt (line 9)) (3.2.2)
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\install.py", line 27, in <module>
    add_global_shortcut_module()
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\install.py", line 13, in add_global_shortcut_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\__init__.py", line 1, in <module>
    from . import canny, hed, midas, mlsd, openpose, uniformer
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\canny\__init__.py", line 1, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\install.py module for custom nodes: No module named 'cv2'Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\midas\__init__.py", line 1, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\midas module for custom nodes: No module named 'cv2'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\mlsd\__init__.py", line 1, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\mlsd module for custom nodes: No module named 'cv2'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\openpose\__init__.py", line 6, in <module>
    from . import util
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\openpose\util.py", line 3, in <module>
    import matplotlib
ModuleNotFoundError: No module named 'matplotlib'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\openpose module for custom nodes: No module named 'matplotlib'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\uniformer\__init__.py", line 3, in <module>
    from comfy_controlnet_preprocessors.uniformer.mmseg.apis import init_segmentor, inference_segmentor, show_result_pyplot
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\uniformer\__init__.py", line 3, in <module>
    from comfy_controlnet_preprocessors.uniformer.mmseg.apis import init_segmentor, inference_segmentor, show_result_pyplot
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\uniformer\mmseg\apis\__init__.py", line 1, in <module>
    from .inference import inference_segmentor, init_segmentor, show_result_pyplot
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\uniformer\mmseg\apis\inference.py", line 1, in <module>
    import matplotlib.pyplot as plt
ModuleNotFoundError: No module named 'matplotlib'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\uniformer module for custom nodes: No module named 'matplotlib'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\util.py", line 2, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\util.py module for custom nodes: No module named 'cv2'
Traceback (most recent call last):
  File "D:\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1003, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\__init__.py", line 1, in <module>
    from . import canny, hed, midas, mlsd, openpose, uniformer
  File "D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\canny\__init__.py", line 1, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'

Cannot import D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\__init__.py module for custom nodes: No module named 'cv2'
Starting server

To see the GUI go to: http://127.0.0.1:8188 `

**Comment by Fannovel16** [comment]
Did you run install.py?

**Comment by Fannovel16** [comment]
That is weird. Did you run this command?
```
..\..\..\python_embeded\python.exe -m pip install -r requirements.txt --extra-index-url https://download.pytorch.org/whl/cu117 --no-warn-script-location
```

**Comment by fambaa** [comment]
I just tried again on the standalone version this is the output from the install, some setup.py error


`PS D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_controlnet_preprocessors> ..\..\..\python_embeded\python.exe -m pip install -r requirements.txt --extra-index-url https://download.pytorch.org/whl/cu117 --no-warn-script-location
Looking in indexes: https://pypi.org/simple, https://download.pytorch.org/whl/cu117
Collecting opencv-contrib-python
  Using cached opencv_contrib_python-4.7.0.72-cp37-abi3-win_amd64.whl (44.9 MB)
Collecting opencv-python
  Using cached opencv_python-4.7.0.72-cp37-abi3-win_amd64.whl (38.2 MB)
Collecting numpy==1.23.1
  Using cached numpy-1.23.1-cp310-cp310-win_amd64.whl (14.6 MB)
Requirement already satisfied: timm==0.6.12 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from -r requirements.txt (line 4)) (0.6.12)
Requirement already satisfied: torchvision in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from -r requirements.txt (line 5)) (0.14.1+cu117)
Requirement already satisfied: scipy in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from -r requirements.txt (line 6)) (1.10.1)
Collecting matplotlib
  Using cached matplotlib-3.7.1-cp310-cp310-win_amd64.whl (7.6 MB)
Collecting scikit-image
  Using cached scikit_image-0.20.0-cp310-cp310-win_amd64.whl (23.7 MB)
Collecting basicsr==1.4.2
  Using cached basicsr-1.4.2.tar.gz (172 kB)
  Preparing metadata (setup.py) ... done
Collecting prettytable==3.6.0
  Using cached prettytable-3.6.0-py3-none-any.whl (27 kB)
Collecting svglib
  Using cached svglib-1.5.1-py3-none-any.whl
Collecting reportlab
  Using cached reportlab-3.6.12-cp310-cp310-win_amd64.whl (2.3 MB)
Requirement already satisfied: huggingface-hub in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from timm==0.6.12->-r requirements.txt (line 4)) (0.13.2)
Requirement already satisfied: pyyaml in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from timm==0.6.12->-r requirements.txt (line 4)) (6.0)
Requirement already satisfied: torch>=1.7 in d:\comfyui_windows_portable\python_embeded\lib\site-packages (from timm==0.6.12->-r requirements.txt (line 4)) (1.13.1+cu117)
Collecting addict
  Using cached addict-2.4.0-py3-none-any.whl (3.8 kB)
Collecting future
  Using cached future-0.18.3.tar.gz (840 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\chris\AppData\Local\Temp\pip-install-043kyae6\future_b342c135c64642c7a26256cd7af5a989\setup.py", line 86, in <module>
          import src.future
      ModuleNotFoundError: No module named 'src'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
PS D:\ComfyUI_windows_portable\ComfyUI\custom_nodes\comfy_controlnet_preprocessors>`


Even though here it says it installs madplotlib, when I start comfy with the bat its log looks exactly liek the one I posted.

As I mentioned before, this seems to be some kind of problem with the standalone python.
My normal install with the python I have on my PC runs fine. And as you can see the standalone even knows the PATH and pulls from cached versions on my disk.

my cv2 version on my PC is 4.7 or something.

I am not really familiar with python, maybe it has to do with the python/pip version the standalone comes with?

**Comment by Fannovel16** [comment]
Fixed in this PR: https://github.com/Fannovel16/comfy_controlnet_preprocessors/commit/b4e4493bee1d738e0862a99ce3cba88e29a0c9c0

The problem is that the original ControlNet repo use basicsr to just download checkpoint files. To do so, it will install future package. For some unknown reasons, the python build from the standalone version can't install it. So I simply removed that package and put load_file_from_url's code into util.py
